File: dandanplay/lib/Model/Message/Send/LoadDanmakuMessage.dart

import 'package:dandanplay/Model/Message/Send/BaseMessage.dart';
import 'package:json_annotation/json_annotation.dart';

part 'LoadDanmakuMessage.g.dart';

@JsonSerializable()
class LoadDanmakuMessage extends BaseMessage {

  @JsonKey(ignore: true)
  @override
  String get name => "LoadDanmakuMessage";

  @override
  Map<String, dynamic> get data {
    return _$LoadDanmakuMessageToJson(this);
  }

  String mediaId;
  Map<String, dynamic> danmakuCollection;
  String title;
  int episodeId;
  //立即播放，无视弹幕
  bool playImmediately = false;

  LoadDanmakuMessage({this.mediaId, this.danmakuCollection, this.title, this.playImmediately, this.episodeId});

}

File: CocoaShare/Model/Message/Receive/LoadDanmakuMessage.swift
//
//  LoadDanmakuMessage.swift
//  dandanplay_native
//
//  Created by JimHuang on 2020/2/19.
//

open class LoadDanmakuMessage: BaseModel {
    open var danmakuCollection: DanmakuCollectionModel?
    open var mediaId = ""
    open var title: String?
    open var playImmediately = false
    open var episodeId = 0
}
